nixnet/_utils.py

```diff
@@ -28,23 +29,25 @@ def flatten_items(list):
         # For FRAME_IN_QUEUED / FRAME_OUT_QUEUED
         # Convenience for everything else
         if ',' in list:
-            # A bit of an abuse of an error code
-            _errors.check_for_error(_cconsts.NX_ERR_INVALID_PROPERTY_VALUE)
+            error_code = _cconsts.NX_ERR_INVALID_PROPERTY_VALUE
+            error_text = _errors.status_to_string(error_code)
+            raise errors.XnetError(error_text, error_code)
```

Comment:
Why are you no longer calling `check_for_error`?

Reply:
I see there is a dedicated commit for this but nothing in the message gives an explanation for it.

Reply:
Why do you want `check_for_error` to be called? If I know I want to raise an error, why would I call a method that conditionally raises or warns instead of raising the error directly? Originally, I thought this was the "abuse" you were mentioning in the comments, though given your questioning here I suspect you meant the abuse was that you just picked the best fitting error code. That wasn't clear to me.

Side note: in another PR I have, calling `check_for_error` instead of raising directly confuses mypy into complaining that there is a path without a return value.

Reply:
> Why do you want check_for_error to be called? If I know I want to raise an error, why would I call a method that conditionally raises or warns instead of raising the error directly?

My primary concern is that changes are explained in the commit message.

My (slight) preference for `check_for_error` is that it is a one-stop shop for getting the appropriate behavior in the correct way.  I kept `XnetError`'s `__init__` "low level" for the sake of being easy to create whatever type of `XnetError` we need and assumed everyone would create them via `check_for_error`.

> Originally, I thought this was the "abuse" you were mentioning in the comments, though given your questioning here I suspect you meant the abuse was that you just picked the best fitting error code. That wasn't clear to me.

Yes, it is referring to the error code

> Side note: in another PR I have, calling check_for_error instead of raising directly confuses mypy into complaining that there is a path without a return value.

One way of resolving this is we could add a dependency on `mypy_extensions` and use its [`NoReturn`](https://mypy.readthedocs.io/en/latest/kinds_of_types.html#the-noreturn-type) type annotation.

On the other hand, `check_for_error` is not unconditionally a `NoReturn` due to the warning case.  Hmmm that is one thing in favor of not using `check_for_error`.

Reply:
What would you think about adding a method in _errors.py

```
def raise_error(error_code):
```

that creates XnetError and raises it. Then, those two methods in _errors.py would be our one-stop shop and my hope is that mypy would be smart enough to figure out that a code path involving `raise_error` unconditionally throws.

Reply:
I added a new method `raise_xnet_error` to use when we unconditionally want an error, and which is called by `check_for_error`. This also resolve the mypy return issue.